RFC/src/RFC-0190_Mempool.md

```diff
@@ -0,0 +1,238 @@
+- All Verified transaction MUST be propagated to neighboring peers.
+- Duplicate transactions MUST NOT be propagated to peers.
+- Unvalidated or invalid transactions MUST NOT be propagated to peers.
+- Verified transactions that were discarded due to low priority levels MUST be propagated to peers.
+- The Mempool MUST have an interface that can be used by neighboring peers to query the content and state of the memory 
+pool.
+- It MUST have a mechanism that enables peers to download the memory pool state in full or in part.
+- It MUST accept all transactions received from peers but MAY decide to discard low priority transactions.
+- It MUST allow Wallets to track incoming payments by monitoring new transaction that has been added to the Mempool.
+- A Mempool MAY choose:
+    - to discard the Mempool state on restarts and then download the full state from its peers or
+    - to store the state of the Mempool using persistent storage to reduce communication bandwidth required when 
+    reinitializing the Mempool after a restart.
+
+### Transaction Pool
+
+The Transaction Pool consists of all unconfirmed transactions that have been received, verified and have passed 
+all checks. These unconfirmed transactions in the Transaction Pool are ready to be included and can be used to 
+construct new blocks for the Tari blockchain.
+
+Functional behavior required of the Transaction Pool:
+- It MUST verify that incoming transactions only spend existing UTXOs.
+- It MUST ensure that incoming transactions don't have a processing time-lock or has a time-lock that has 
+expired.
+- It MUST ensure that all time-locks of the UTXOs that will be spent by the transaction have expired.
+- Transactions that have been used to construct new blocks MUST be removed from the Transaction Pool and added to the Reorg Pool.
+
+### Pending Pool
+
+The Pending Pool contains all transactions that are restricted by time-locks. A transaction could have a time-lock 
+limiting it from being processed or it can attempt to spend UTXOs with time-locks. These transactions require 
+their time-locks or the time-locks of the input UTXOs to expire before they can be processed and included into new 
+blocks. All transactions in the Pending Pool have been verified and passed all checks except their own time-lock has 
+not yet expired or some of the UTXOs that will be spent have time-lock restrictions that are not yet valid. Once the 
+transactions time-lock or the time-locks on the UTXOs have expired then the Pending Pool transactions can be moved 
+to the Transaction Pool for inclusion in future blocks.  
+
+Functional behavior required of the Pending Pool:
+- Once the transaction time-lock or UTXO time-lock restricting the processing of a transaction has expired then the 
+pending transaction MUST be moved to the Transaction Pool.
+- Pending transactions that have matured past the discard threshold MUST be discarded and removed from the Pending Pool.
```

Comment:
Only applicable for the reorg pool and maybe the orphan pool?

Reply:
Correct, item was removed